docs/src/tutorials/advanced_simulations.md

```diff
@@ -0,0 +1,215 @@
+```@example ex1
+esol = solve(eprob; trajectories=5)
+```
+This simulation is automatically multithreaded over all available threads. Please read [this documentation](https://docs.sciml.ai/DiffEqDocs/stable/features/ensemble/#EnsembleAlgorithms) for more information on parallelisation alternatives. The ensemble simulations can be plotted using the `plot()` function, which will by default display all trajectories:
+```@example ex1
+plot(esol)
+```
+
+Sometimes when performing a large number of ensemble simulations, the plots get very dense. In these cases, the plotting argument is `linealpha` (which sets trajectory transparency) may be useful:
+```@example ex1
+esol = solve(eprob; trajectories=100)
+plot(esol)
+```
+
+Sometimes, one wishes to perform the same simulation a large number of times, while making minor modifications to the problem each time. This can be done by giving a problem function argument to the `EnsembleProblem`. Let us consider ODE simulations of a simple birth/death process:
+```@example ex1
+rn = @reaction_network begin
+    (b,1.0), ∅ <--> X
+end
+u0 = [:X => 1.0]
+tspan = (0.0,1.0)
+p = [:b => 1.];
+oprob = ODEProblem(rn,u0,tspan,p)
+nothing # hide
+```
+We wish to simulate this model for a large number of values of `b`. We do this by creating a `prob_func` that will make a modification to the problem at the start of each Monte Carlo simulation:
+```@example ex1
+b_values = 1.0:0.1:2.0
+function prob_func(prob,i,repeat)
+    remake(prob; p=[b_values[i]])
+end
+nothing # hide
+```
+Here, `prob_func` takes three arguments:
+ - `prob`: The problem given to our `EnsembleProblem`, this is the problem that `prob_func` modified in each iteration.
+ - `i`: The number of this specific Monte Carlo iteration in the interval `1:trajectories`.
+ - `repeat`: The repeat of this specific Monte Carlo simulation (We will ignore this argument in this brief overview).
+In our case, for each Monte Carlo simulation, our `prob_func` takes our original `ODEProblem` and uses the `remake` function to change the parameter vector. Here, for the `i`th Monte Carlo simulation, the value of `b` is also the `i`th value of our `b_values` vector. Finally, we can simulate and plot our problem:
+```@example ex1
+eprob = EnsembleProblem(oprob; prob_func=prob_func)
+esol = solve(eprob; trajectories=length(b_values))
+plot(esol)
+```
+
+Finally, it is worth noting that plot legends are disabled when plotting ensemble solutions. These can be re-enabled using the `legend` plotting keyword. However, when plotting a large number of trajectories, each will generate a label. Sometimes the best approach is to remove these and add a label manually:
+```
+plot(esol; label="")
+plot!([],[],label="X", legend=:best)
+```
+
+
+### Event handling using callbacks
```

Comment:
Before we release Catalyst 13 this should be modified to use the symbolic event interface via `ReactionSystem`s (I'll add discrete event support).

Reply:
Yes, I kind of left this lose for now since there are currently several MTK updates underway that might change how we do this.